modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/MatchOnlyTextFieldMapper.java

```diff
@@ -249,59 +244,31 @@ private IOFunction<LeafReaderContext, CheckedIntFunction<List<Object>, IOExcepti
                     "Field [" + name() + "] of type [" + CONTENT_TYPE + "] cannot run positional queries since [_source] is disabled."
                 );
             }
-            if (searchExecutionContext.isSourceSynthetic() && withinMultiField) {
```

Comment:
this block of code was broken down into three smaller functions for readability:
- `parentFieldValueFetcher`
- `delegateFieldValueFetcher`
- `storedFieldValueFetcher`